packages/turbo-types/src/types/config.ts

```diff
@@ -3,6 +3,15 @@ export interface Schema {
   /** @default https://turbo.build/schema.json */
   $schema?: string;
 
+  /**
+   * A version of turbo that should be used with this project
+   *
+   * If turbo is invoked in a project with turboVersion specified, turbo will make
+   * sure to use the correct version. Only semver versions that include patch
+   * numbers are allowed.
```

Comment:
- Are canaries allowed? 
- One of the things I really dislike about corepack is that I have to choose an exact version. Most of the time, I don't know what minor or patch versions are even available, and I can never remember how/where to find that info. As a maintainer of turbo, I'd want exact version also, but as a user, I would really just want `1.x` or at most `1.5.x`

Reply:
> Are canaries allowed?

[Yup](https://github.com/Masterminds/semver/blob/v3.1.1/version.go#L195)! Prereleases and build metadata is allowed.

> I would really just want 1.x or at most 1.5.x

It's annoying, but I really think we need to be strict here. Say `turboVersion` is `1.5.x` and Dev 1 gets a failure due to a bug in turbo `1.5.0`, Dev 2 tries to help so they clone the repo and run the same command, but we've pushed a patch that solves it so they don't see it and causes more confusion. The update notifier will help with this scenario, but it still means that a team can have version drift.

I think the better solution would be to have a codemod available that can resolve a version constraint or tag to a concrete version. e.g. `npx @turbo/codemod upgrade-turbo 1.5`